pilot/pkg/controllers/ipallocate/ipallocate.go

```diff
@@ -0,0 +1,413 @@
+				continue
+			}
+			c.markUsedOrQueueConflict(a, se)
+		}
+		return nil // nothing to do
+	}
+
+	patch, err := c.statusPatchForAddresses(serviceentry, false)
+	if err != nil {
+		return err
+	}
+
+	if patch == nil {
+		return nil // nothing to patch
+	}
+
+	_, err = c.serviceEntryClient.PatchStatus(se.Name, se.Namespace, types.MergePatchType, patch)
+	if err != nil {
+		log.Errorf("unable to patch %s, received error %s", se.String(), err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (c *IPAllocate) resolveConflict(conflict conflictDetectedEvent) error {
+	log.Errorf("I guess I should resolve this %v", conflict)
+	var autoConflicts, userConflicts []*networkingv1alpha3.ServiceEntry
+	serviceentries := c.serviceEntryClient.List(metav1.NamespaceAll, klabels.Everything())
+	for _, serviceentry := range serviceentries {
+		auto, user := allAddresses(serviceentry)
+		if slices.Contains(auto, conflict.conflictingAddress) {
+			autoConflicts = append(autoConflicts, serviceentry)
+		}
+		if slices.Contains(user, conflict.conflictingAddress) {
+			userConflicts = append(userConflicts, serviceentry)
+		}
+	}
+
+	slices.SortFunc(autoConflicts, func(a, b *networkingv1alpha3.ServiceEntry) int {
+		return a.CreationTimestamp.Compare(b.CreationTimestamp.Time)
+	})
+
+	start := 1
+
+	if len(userConflicts) > 0 {
+		start = 0
+	}
+	var errs error
+	for _, resolveMe := range autoConflicts[start:] {
+		log.Errorf("I should re-assign %s/%s", resolveMe.Namespace, resolveMe.Name)
+		patch, err := c.statusPatchForAddresses(resolveMe, true)
+		if err != nil {
+			errs = errors.Join(errs, err)
+		}
+
+		if patch == nil {
+			errs = errors.Join(errs, fmt.Errorf("this should not occur but patch was empty on a forced reassignment"))
+		}
+
+		_, err = c.serviceEntryClient.PatchStatus(resolveMe.Name, resolveMe.Namespace, types.MergePatchType, patch)
+		if err != nil {
+			log.Errorf("unable to patch %s/%s, received error %s", resolveMe.Namespace, resolveMe.Name, err.Error())
+			errs = errors.Join(errs, err)
+		}
+	}
+
+	return errs
+}
+
+func allAddresses(se *networkingv1alpha3.ServiceEntry) ([]netip.Addr, []netip.Addr) {
+	if se == nil {
+		return nil, nil
+	}
+	autoAssigned := autoallocate.GetV2AddressesFromServiceEntry(se)
+	userAssigned := []netip.Addr{}
+
+	for _, a := range se.Spec.Addresses {
+		parsed, err := netip.ParseAddr(a)
+		if err != nil {
+			// maybe log
+			continue
+		}
+		userAssigned = append(userAssigned, parsed)
+	}
+	return autoAssigned, userAssigned
+}
+
+func (c *IPAllocate) nextAddresses(owner types.NamespacedName) []netip.Addr {
+	v4address, err := c.v4allocator.AllocateNext(owner)
+	if err != nil {
+		panic("failed to allocate v4 address")
+	}
+	v6address, err := c.v6allocator.AllocateNext(owner)
+	if err != nil {
+		panic("failed to allocate v6 address")
+	}
+	return []netip.Addr{
+		v4address,
+		v6address,
+	}
+}
+
+func (c *IPAllocate) inOurRange(a netip.Addr) bool {
+	if a.Is6() {
+		return c.v6allocator.prefix.Contains(a)
```

Comment:
IIRC this will return `true` for 4in6 addresses? so we might need to unwrap before checking prefix?

Reply:
good catch, thankyou